pep-0277.txt

```diff
@@ -5,114 +5,126 @@ Last-Modified: $Date$
+the other argument converted to Unicode using the default
+encoding.
+
+The ``listdir`` function currently returns a list of strings.  Under
+this proposal, it will return a list of Unicode strings when its
+path argument is Unicode.
 
 
 Restrictions
-
-    On the consumer Windows operating systems, Windows 95, Windows 98,
-    and Windows ME, there are no wide-character file APIs so behaviour
-    is unchanged under this proposal.  It may be possible in the
-    future to extend this proposal to cover these operating systems as
-    the VFAT-32 file system used by them does support Unicode file
-    names but access is difficult and so implementing this would
-    require much work.  The "Microsoft Layer for Unicode" could be a
-    starting point for implementing this.
-
-    Python can be compiled with the size of Unicode characters set to
-    4 bytes rather than 2 by defining PY_UNICODE_TYPE to be a 4 byte
-    type and Py_UNICODE_SIZE to be 4.  As the Windows API does not
-    accept 4 byte characters, the features described in this proposal
-    will not work in this mode so the implementation falls back to the
-    current 'mbcs' encoding technique. This restriction could be lifted
-    in the future by performing extra conversions using
-    PyUnicode_AsWideChar but for now that would add too much
-    complexity for a very rarely used feature.
+============
+
+On the consumer Windows operating systems, Windows 95, Windows 98,
+and Windows ME, there are no wide-character file APIs so behaviour
+is unchanged under this proposal.  It may be possible in the
+future to extend this proposal to cover these operating systems as
+the VFAT-32 file system used by them does support Unicode file
+names but access is difficult and so implementing this would
+require much work.  The "Microsoft Layer for Unicode" could be a
+starting point for implementing this.
+
+Python can be compiled with the size of Unicode characters set to
+4 bytes rather than 2 by defining ``PY_UNICODE_TYPE`` to be a 4 byte
+type and ``Py_UNICODE_SIZE`` to be 4.  As the Windows API does not
+accept 4 byte characters, the features described in this proposal
+will not work in this mode so the implementation falls back to the
+current 'mbcs' encoding technique. This restriction could be lifted
+in the future by performing extra conversions using
+``PyUnicode_AsWideChar`` but for now that would add too much
+complexity for a very rarely used feature.
 
 
 Reference Implementation
+========================
 
-    An experimental implementation is available from
-    [2] http://scintilla.sourceforge.net/winunichanges.zip
+An experimental implementation is available from [2]_.
+
+An updated version is available at [3]_.
 
-    [3] An updated version is available at
-        http://python.org/sf/594001
 
 
 References
+==========
+
+.. [1] Microsoft Windows APIs
+       http://msdn.microsoft.com/
+
+.. [2] http://scintilla.sourceforge.net/winunichanges.zip
```

Comment:
Since the final patch has been attached to bugs.python.org, we can now delete the sourceforge.net link.